### PR TITLE
Updated CI environment

### PR DESCRIPTION
**Pull request by artemry-mlnx**
- Updated after CI cluster upgrade
- Aligned with the mainline infra

### CODE REVIEW

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 30-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/61e8058a8740ea19458c91e99e4407dc/raw/5406d3cbd9296461ac63db1961ea21c383d85cbb/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/61e8058a8740ea19458c91e99e4407dc/raw/9aa65743042c3aa802eb204a37a142787f9caaf8/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/61e8058a8740ea19458c91e99e4407dc/raw/b5693ff478966ad549243f63836dfb5ce91fa268/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/61e8058a8740ea19458c91e99e4407dc/raw/164fae1fd3b1a8d9dff1491a18ac72a66fe188f3/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 30-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/1c08873a759791563898545f8321923c/raw/3c58ed17eac53e61f5f6843e8e4ef5c14f0f3c30/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/1c08873a759791563898545f8321923c/raw/c194d1d059bdf218672e36564c4917b0481fb538/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/1c08873a759791563898545f8321923c/raw/150deea3a99b7a187f515051a49d8e18bed92531/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/1c08873a759791563898545f8321923c/raw/2c7357c92aefd11256931fe6aab1b8b4a98485d2/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by artemry-mlnx** [comment]
@amaslenn @bureddy could you please review?

CC @Sergei-Lebedev

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 30-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/801d8422f4a5bb56a264e1cba094c4a0/raw/4f32ab7bd37063298c729e8f985ca0d2f00fe60d/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/801d8422f4a5bb56a264e1cba094c4a0/raw/459e8b271a667ae8b3aea17540a8517d24b57fce/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/801d8422f4a5bb56a264e1cba094c4a0/raw/91064c3b6812828a8efb66e38f9fd028fb57c7ee/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/801d8422f4a5bb56a264e1cba094c4a0/raw/2e0db6867906b6053ee00acea1c02931b5a80b0b/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -36,13 +36,19 @@ pipeline {
         stage('Build nccl-rdma-sharp-plugins') {
             steps {
                 echo 'Building nccl-rdma-sharp-plugins...'
-                sh 'set -o pipefail; ${WORKSPACE}/.ci/build_nccl_rdma_sharp_plugins.sh 2>&1 | tee ${ARTEFACT_DIR}/build_nccl_rdma_sharp_plugins.log'
+                sh """#!/bin/bash
```
Comment (amaslenn) [inline .ci/Jenkinsfile]: Don't you need `-eE`? Or maybe also `-l`?

File: .ci/build_nccl_rdma_sharp_plugins.sh [hunk 2]
```diff
@@ -1,42 +1,37 @@
...
-then
+if ! make -j install; then
     echo "ERROR: 'make -j install' failed"
     echo "FAIL"
     exit 1
 fi
 
 echo "INFO: ${NCCL_RDMA_SHARP_PLUGINS_DIR}:"
-find ${NCCL_RDMA_SHARP_PLUGINS_DIR} -type f
+find "${NCCL_RDMA_SHARP_PLUGINS_DIR}" -type f
```
Comment (amaslenn) [inline .ci/build_nccl_rdma_sharp_plugins.sh]: What is the purpose of this line?

File: .ci/settings.sh [hunk 3]
```diff
@@ -12,12 +10,23 @@ fi
 export CUDA_VER="${CUDA_VER:-10.2}"
 echo "INFO: CUDA_VER = ${CUDA_VER}"
 
-module load dev/cuda${CUDA_VER}
-module load hpcx-gcc
 module load ml/ci-tools
 
+# W/A for SHARP
+module load dev/cuda10.1
```
Comment (amaslenn) [inline .ci/settings.sh]: Could you please explain how this WA works? Loading two different versions of CUDA (or maybe the same) looks confusing.

File: .ci/build_nccl_rdma_sharp_plugins.sh (hunk 2 above)
Comment (artemry-mlnx) [reply .ci/build_nccl_rdma_sharp_plugins.sh]: Just for debug purposes to see the content of ${NCCL_RDMA_SHARP_PLUGINS_DIR} folder (built binaries).

File: .ci/settings.sh (hunk 3 above)
Comment (artemry-mlnx) [reply .ci/settings.sh]: CUDA 10.2 is the latest available version we'd like to test, CUDA 10.1 is needed for SHARP. CUDA 10.2 has priority in the env PATH/LD_LIBRARY_PATH.

File: .ci/Jenkinsfile (hunk 1 above)
Comment (artemry-mlnx) [reply .ci/Jenkinsfile]: `set -o pipefail` is needed specifically for `... | tee ...` to correctly catch script exit code in this shell pipe. Others are unnecessary here (but we may use them inside the scripts we call during Jenkins step.)

File: .ci/build_nccl_rdma_sharp_plugins.sh (hunk 2 above)
Comment (amaslenn) [reply .ci/build_nccl_rdma_sharp_plugins.sh]: Could you please add this comment to the script?

File: .ci/settings.sh (hunk 3 above)
Comment (amaslenn) [reply .ci/settings.sh]: Could you please add this comment to the script?

Comment (artemry-mlnx) [reply .ci/settings.sh]: Fixed.

File: .ci/build_nccl_rdma_sharp_plugins.sh (hunk 2 above)
Comment (artemry-mlnx) [reply .ci/build_nccl_rdma_sharp_plugins.sh]: Fixed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 31-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/f1d8a0a9ec0a0d7394ed76efc89fcdfc/raw/49b43f70c304142592049afb741d72048ad9c8f1/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/f1d8a0a9ec0a0d7394ed76efc89fcdfc/raw/0fdb7b042e0e30b9699f17cc2248c3c2d2bfc8d4/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/f1d8a0a9ec0a0d7394ed76efc89fcdfc/raw/ac96c6d16428948f6695f6e0454f49198c02ac23/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/f1d8a0a9ec0a0d7394ed76efc89fcdfc/raw/698f0678a24fa7622b1f94c067cdb47e2acf701c/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 31-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/e6132ce16ace3862b0fe35bb6bf2c6d2/raw/ebde4c68fe798ca87574e4afee36e9125623a1c0/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/e6132ce16ace3862b0fe35bb6bf2c6d2/raw/b3cb9ac3150fce05ad60b4727ee323434b946ced/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/e6132ce16ace3862b0fe35bb6bf2c6d2/raw/8b51e11b9eb930c08cdb89debddb5f6e58b2d729/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/e6132ce16ace3862b0fe35bb6bf2c6d2/raw/7b8576364fbd934ad11c0b71e89d913088d3b135/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**